python_modules/dagster/dagster/components/core/load_defs.py

```diff
@@ -81,9 +84,10 @@ def load_defs(defs_root: ModuleType, project_root: Optional[Path] = None) -> Def
     # create a top-level DefsModule component from the root module
     context = ComponentLoadContext.for_module(defs_root, project_root)
     with use_component_load_context(context):
-        root_component = get_component(context)
+        hierarchy = build_component_hierarchy(context)
+        root_component = build_root_component(context, hierarchy)
```

Comment:
i think if were going to have two trees, we will want a wrapper object for the final output too that has a reference to both trees. Not sure we need to resolve that fully in this PR

Reply:
yeah would rather wait for concrete need for usage to assist with naming